app/assets/javascripts/initializers/initScrolling.js.erb

```diff
@@ -24,21 +24,38 @@ function checkIfNearBottomOfPage() {
 }
 
 function fetchNextPageIfNearBottom() {
-  var elCheck = document.getElementById("index-container") && !document.getElementById("query-wrapper");
+  var indexContainer = document.getElementById("index-container");
+  var elCheck =  indexContainer && !document.getElementById("query-wrapper");
   if (!elCheck) {
     return;
   }
-  if ( !done && !fetching && (window.scrollY) > (document.documentElement.scrollHeight - 3700) ) {
+  
+  var indexWhich = indexContainer.dataset.which;
+
+  var fetchCallback;
+  var scrollableElemId;
+  if (indexWhich == "podcast-episodes") {
+    scrollableElemId = "articles-list";
+    fetchCallback = function() {
+      fetchNextPodcastPage(indexContainer);
+    };
+  } else if (indexWhich == "videos") {
+    scrollableElemId = "video-collection";
+    fetchCallback = function() {
+      fetchNextVideoPage(indexContainer);
+    };
+  } else {
+    scrollableElemId = "articles-list";
+    fetchCallback = function() {
+      algoliaPaginate(indexContainer.dataset.algoliaTag);
+    };
+  }
+
+  var scrollableElem = document.getElementById(scrollableElemId);
+
+  if ( !done && !fetching && (window.scrollY) > (scrollableElem.scrollHeight - 3700) ) {
```

Comment:
Previously, `document.documentElement` was used in this calculation, instead of the element we actually wanted to scroll.

When the additional sidebar on the right reaches a certain threshold in height, it interferes with the scroll height calculation when using `document.documentElement`.

The fix here is to determine where `window.scrollY` is in comparison to the element we care about scrolling, `scrollableElem`.